### PR TITLE
Create works_management app

### PR DESCRIPTION
**Pull request by kdeloach**
Create Django app, models, migrations, and unit tests for
works_management feature.

Connects #2996

### CODE REVIEW

**Comment by jwalgran** [comment]
There is some lint

```
$ ./scripts/check.sh
+ vagrant ssh app -c 'cd /opt/app/core/opentreemap; flake8 --exclude migrations,opentreemap/settings/local_settings.py *'
works_management/urls.py:1:1: F401 'django.conf.urls.include' imported but unused
works_management/urls.py:1:1: F401 'django.conf.urls.url' imported but unused
Connection to 127.0.0.1 closed.

File: opentreemap/works_management/tests.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def make_plot(instance, user):
+    p = Point(0, 0)
+    plot = Plot(geom=p, instance=instance)
+    plot.save_with_user(user)
+    return plot
+
+
+class WorksManagementTests(OTMTestCase):
```
Comment (jwalgran) [inline opentreemap/works_management/tests.py]: I would like to add these two tests.

```python
    def test_cannot_call_save_on_workorder(self):
        # WorkOrder is Auditible and should only allow calling save_with_user.
        w = make_work_order(self.instance, self.user)
        with self.assertRaises(UserTrackingException):
            w.save()

    def test_cannot_call_save_on_task(self):
        # Task is Auditible and should only allow calling save_with_user.
        t = make_task(self.instance, self.user,
                      make_plot(self.instance, self.user))
        with self.assertRaises(UserTrackingException):
            t.save()
```

The WorkOrder test fails. I would expect `save` function in Auditible to be called, but it is not. 

https://github.com/OpenTreeMap/otm-core/blob/746322311ef4df3e550dfdcfd81f9848a5202c53/opentreemap/treemap/audit.py#L615-L618

If we cannot figure out the inheritance issue, the simple solution is to copy the `save` function into the WorkOrder model.

```
    def save(self, *args, **kwargs):
        raise UserTrackingException(
            'All changes to %s objects must be saved via "save_with_user"' %
            (self._model_name))
```

File: opentreemap/works_management/models.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+class Team(models.Model):
+    instance = models.ForeignKey(Instance)
+    name = models.CharField(max_length=255)
+
+
+class WorkOrder(models.Model, Auditable):
+    instance = models.ForeignKey(Instance)
+    name = models.CharField(max_length=255)
```
Comment (kdeloach) [inline opentreemap/works_management/models.py]: It should be non-nullable by default, right? But I'll set `null=False` and `blank=False` just to be safe.

File: opentreemap/works_management/models.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        (COMPLETED, _('Completed')),
+        (CANCELED, _('Canceled')),
+    )
+
+    instance = models.ForeignKey(Instance)
+    map_feature = models.ForeignKey(MapFeature)
+    work_order = models.ForeignKey(WorkOrder, null=True, related_name='tasks')
+    team = models.ForeignKey(Team, null=True)
+
+    name = models.CharField(max_length=255)
```
Comment (kdeloach) [inline opentreemap/works_management/models.py]: Thanks, must have been a copy & paste error.

File: opentreemap/works_management/models.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        },
+    }
+
+    def save_with_user(self, user, *args, **kwargs):
+        """
+        Update WorkOrder fields when Task is saved.
+        """
+        if self.work_order:
+            self.work_order.updated_at = timezone.now()
+            self.work_order.save()
```
Comment (kdeloach) [inline opentreemap/works_management/models.py]: I wonder why this doesn't trigger an exception in the unit test? I'll investigate.

File: opentreemap/works_management/tests.py (hunk 1 above)
Comment (kdeloach) [inline opentreemap/works_management/tests.py]: Good call. I was just thinking the same thing after your last comment.

Comment (kdeloach) [inline opentreemap/works_management/tests.py]: The solution is to make sure we inherit from Auditable before Model:

```
-class WorkOrder(models.Model, Auditable):
+class WorkOrder(Auditable, models.Model):
```

**Comment by kdeloach** [comment]
Updates:

* Fix lint
* Fix `Auditable` inheritance issue on `WorkOrder` model
* Unit tests to make sure `save_with_user` is enforced

File: opentreemap/works_management/models.py (hunk 2 above)
Comment (jwalgran) [inline opentreemap/works_management/models.py]: :man_facepalming: I should not comment before sufficient caffeine.

**Comment by kdeloach** [comment]
Thanks.

**Pull request merged**